samples/guide/src/main/java/okhttp3/recipes/kt/WiresharkExample.kt

```diff
@@ -186,37 +243,30 @@ class WireSharkKeyLoggerListener(
       logger = Logger.getLogger("javax.net.ssl")
           .apply {
             level = Level.FINEST
+            useParentHandlers = false
           }
     }
   }
 }
 
-val eventListenerFactory = WireSharkKeyLoggerListener.Factory(File("/tmp/key.log"), verbose = true)
-
 @SuppressSignatureCheck
-class WiresharkExample(private val tlsVersions: List<TlsVersion>) {
-  val connectionSpec =
+class WiresharkExample(tlsVersions: List<TlsVersion>, private val launch: Launch? = null) {
+  private val connectionSpec =
     ConnectionSpec.Builder(ConnectionSpec.RESTRICTED_TLS)
         .tlsVersions(*tlsVersions.toTypedArray())
         .build()
 
+  private val eventListenerFactory = WireSharkListenerFactory(
+      logFile = File("/tmp/key.log"), tlsVersions = tlsVersions, launch = launch)
```

Comment:
Passing tlsVersions here is a bit clumsy... But makes the copy and paste keep the warning messages on.  I could react to the negotiated connection instead.